platform/linux-generic/arch/aarch64/odp_crypto_armv8.c

```diff
@@ -449,6 +476,261 @@ static int process_aes_gcm_param(odp_crypto_generic_session_t *session)
+	ODP_ASSERT(session->p.cipher_iv_len == 0 || iv_ptr != NULL);
+#endif
+
+	armv8_expandkeys_dec_aes_cbc_128(key_expanded, session->cipher.key_data);
+	arg.cipher.key = key_expanded;
+	arg.cipher.iv = iv_ptr;
+
+	arg.digest.hmac.key = session->auth.key;
+	arg.digest.hmac.i_key_pad = session->auth.i_key_pad;
+	arg.digest.hmac.o_key_pad = session->auth.o_key_pad;
+
+	uint32_t seg_len = 0;
+	uint8_t *data = odp_packet_offset(pkt, in_pos, &seg_len, NULL);
+
+	if (odp_unlikely(odp_packet_is_segmented(pkt)) ||
+	    odp_unlikely(odp_packet_tailroom(pkt) < OOB_WRITE_LEN)) {
+		/* Packet is segmented or it may not be safe to read and write
+		 * beyond the end of packet data. Copy the cipher range to a
+		 * contiguous buffer. */
+		odp_packet_copy_to_mem(pkt, in_pos, in_len, local.buffer);
+
+		data = local.buffer;
+	}
+
+	seg_len = 0;
+	uint8_t *hash = odp_packet_offset(pkt, hash_pos, &seg_len, NULL);
+
+	if (odp_unlikely(odp_packet_is_segmented(pkt)) ||
+	    odp_unlikely(odp_packet_tailroom(pkt) < OOB_WRITE_LEN)) {
+		/* Packet is segmented or it may not be safe to read and write
+		 * beyond the end of packet data. Copy the cipher range to a
+		 * contiguous buffer. */
+		odp_packet_copy_to_mem(pkt, hash_pos, hash_len, hash_buf);
+
+		hash = hash_buf;
+	}
+
+	if (odp_unlikely(session->p.hash_result_in_auth_range))
+		_odp_packet_set_data(pkt, param->hash_result_offset, 0, session->p.auth_digest_len);
+
+	switch (session->p.auth_alg) {
+	case ODP_AUTH_ALG_SHA1_HMAC:
+		rc = armv8_dec_aes_cbc_sha1_128(data, local.buffer, in_len,
+						hash, digest, hash_len,
+						&arg);
+
+		if (odp_unlikely(rc)) {
+			ODP_DBG("ARM Crypto: AES-CBC Decoding and HMAC-SHA1 "
+				"digest generation failed\n");
+			goto err;
+		}
+		break;
+	case ODP_AUTH_ALG_SHA256_HMAC:
+		rc = armv8_dec_aes_cbc_sha256_128(data, local.buffer, in_len,
+						  hash, digest, hash_len,
+						  &arg);
+
+		if (odp_unlikely(rc)) {
+			ODP_DBG("ARM Crypto: AES-CBC Decoding and HMAC-SHA256 "
+				"digest generation failed\n");
+			goto err;
+		}
+		break;
+	default:
+		goto err;
+	}
+
+	odp_packet_copy_from_mem(pkt, in_pos, in_len, local.buffer);
+
+	set_crypto_op_result_ok(pkt);
+	return;
+
+err:
+	set_crypto_op_result(pkt,
+			     ODP_CRYPTO_ALG_ERR_NONE,
+			     ODP_CRYPTO_ALG_ERR_ICV_CHECK);
+}
+
+static int process_cipher_auth_hmac_param(odp_crypto_generic_session_t *session)
+{
+	/* Verify cipher Key len is valid */
+	if (16 != session->p.cipher_key.length)
+		return -1;
+
+	/* Verify cipher IV len is correct */
+	if (ARM_CRYPTO_MAX_IV_LENGTH != session->p.cipher_iv_len)
+		return -1;
+
+	/* Verify auth IV len is correct */
+	if (0 != session->p.auth_iv_len)
+		return -1;
+
+	/* Zero memory under key */
+	if (session->p.auth_alg == ODP_AUTH_ALG_SHA1_HMAC)
+		memset(session->auth.key, 0, sizeof(session->auth.key));
+	else if (session->p.auth_alg == ODP_AUTH_ALG_SHA256_HMAC)
+		memset(session->auth.key, 0, sizeof(session->auth.key));
+
+	/* Convert keys */
```

Comment:
This is not converting anything but just copying the keys.

Actually, I think the comments in this function do not add much value and could be just removed.